src/Swift.Bindings/src/Emitter/StringEmitter/Handler/TypeHandler.cs

```diff
@@ -519,16 +519,136 @@ IntPtr ISwiftObject.MarshalToSwift(IntPtr swiftDest)
         /// </summary>
         private void WriteGetProtocolConformanceDescriptor()
         {
+            WriteStaticConstructor();
+            var libPath = _typeDatabase.GetLibraryPath(_moduleDecl.Name);
             var text = $$"""
-            static ProtocolConformanceDescriptor ISwiftObject.GetProtocolConformanceDescriptor<U>()
+            static ProtocolConformanceDescriptor ISwiftObject.GetProtocolConformanceDescriptor<TProtocol>()
+                where TProtocol : class
             {
-                throw new NotImplementedException();
+                if (!_protocolConformanceSymbols.TryGetValue(typeof(TProtocol), out var symbolName))
+                {
+                    throw new SwiftRuntimeException($"Attempted to retrieve protocol conformance descriptor for type {{_structDecl.Name}} and protocol {typeof(TProtocol).Name}, but no conformance was found.");
+                }
+
+                return ProtocolConformanceDescriptor.LoadFromSymbol("{{libPath}}", symbolName);
```

Comment:
Do I understand it correctly that we're planning to read symbols from dylibs during runtime? I'm wondering from the security standpoint what constrains we should put on the read dylibs to unsure that we're handling correctly untrusted inputs.

Reply:
My understanding is that we can safely use `MachO` to retrieve the symbols during projection generation - as this doesn't "load' the dylib. However, we cannot load the values behind those symbols until runtime due to potential OS difference. 

I'm not sure how this differs from pinvokes where we do the same thing (compute symbol during projection tooling and call into dylib during runtime)